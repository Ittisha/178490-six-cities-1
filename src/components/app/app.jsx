import * as React from 'react';
import PropTypes from 'prop-types';
import {Switch, Route, Redirect} from 'react-router-dom';

import {Main} from '../main/main';
import Header from '../header/header';
import SignIn from '../sign-in/sign-in';
import Favorites from '../favorites/favorites';
import OfferPage from '../offer-page/offer-page';
import {withFormSubmit} from '../../hoc/with-form-submit/with-form-submit';
import {withPrivateRoutes} from '../../hoc/with-private-routes/with-private-routes';
import {withActiveItem} from '../../hoc/with-active-item/with-active-item';
import {withSortedItems} from '../../hoc/with-sorted-items/with-sorted-items';
import {
  getCities,
  getCitiesCoords,
  getCitiesZoom,
  getCity,
  getCityOffers,
  getIsAuthorized,
} from '../../reducers/selectors';
import {ActionCreator as CitiesActionCreator} from '../../reducers/cities/cities';
import {connect} from 'react-redux';
import {Operation as UserOperation} from '../../reducers/user/user';
import apartmentPropsShape from '../../props/apartment';
import cityPropsShape from '../../props/city';

const SignInWithAuthorization = withFormSubmit(SignIn);
const FavouritesWithPrivateRoutes = withPrivateRoutes(Favorites);
const MainWithActiveItem = withActiveItem(withSortedItems(Main));

class App extends React.PureComponent {
  render() {
    const {

      city,
      cities,
      handleCityChange,
      citiesCoords,
      citiesZoom,
      isAuthorized,
    } = this.props;

    const MainWithPropsAndActiveItem = () => {
      return (<MainWithActiveItem
        city={city}
        cities={cities}
        handleCityChange={handleCityChange}
        citiesCoords={citiesCoords}
        citiesZoom={citiesZoom}
      />
      );
    };

    const redirectSignInPage = () => isAuthorized
      ? <Redirect to="/" />
      : <SignInWithAuthorization />;

    return (
      <React.Fragment>
        <Header />
        <Switch>
          <Route path="/" exact component={MainWithPropsAndActiveItem} />
          <Route path="/login" render={redirectSignInPage} />
          <Route path="/favorites" component={FavouritesWithPrivateRoutes}/>
          <Route path="/offer/:id" component={OfferPage}/>
        </Switch>
      </React.Fragment>
    );
  }
}

App.propTypes = {
  apartments: PropTypes.arrayOf(apartmentPropsShape).isRequired,
  city: cityPropsShape.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  citiesCoords: PropTypes.objectOf(PropTypes.arrayOf(PropTypes.number)).isRequired,
  citiesZoom: PropTypes.objectOf(PropTypes.number).isRequired,
  handleCityChange: PropTypes.func.isRequired,
  isAuthorized: PropTypes.bool.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  city: getCity(state),
  cities: getCities(state),
  citiesCoords: getCitiesCoords(state),
  citiesZoom: getCitiesZoom(state),
  apartments: getCityOffers(state),
  isAuthorized: getIsAuthorized(state),
});

const mapDispatchToProps = (dispatch) => ({
  handleCityChange: (city) => dispatch(CitiesActionCreator.changeCity(city)),
  checkAuthorization: () => dispatch(UserOperation.checkAuthorization()),
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
