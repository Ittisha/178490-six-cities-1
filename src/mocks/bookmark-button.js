export const MOCK_PROPS = {
  offerId: 1,
  isInBookmarks: false,
  className: ``,
  width: 15,
  height: 12,
};
