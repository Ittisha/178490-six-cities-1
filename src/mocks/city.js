export const INIT_CITY = {
  name: `Amsterdam`,
  coords: [52.38333, 4.9],
  zoom: 12,
};

export const MOCK_CITIES = [`Amsterdam`, `Paris`];

export const MOCK_CITIES_COORDS = {
  Amsterdam: [52.38333, 4.9],
  Paris: [48.864716, 2.349014],
};

export const MOCK_CITIES_ZOOM = {
  Amsterdam: 17,
  Paris: 15,
};
