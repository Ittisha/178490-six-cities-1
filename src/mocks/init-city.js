export const INIT_CITY = {
  name: `Amsterdam`,
  coords: [52.38333, 4.9],
};
