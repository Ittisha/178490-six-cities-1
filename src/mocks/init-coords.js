export const INIT_COORDS = [52.38333, 4.9];
