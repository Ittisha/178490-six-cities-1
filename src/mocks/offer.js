export const OFFER = {
  "city": {
    "name": `Cologne`,
    "location": {
      "latitude": 50.938361,
      "longitude": 6.959974,
      "zoom": 13}},
  "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
  "images": [
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": false,
  "rating": 2.5,
  "type": `house`,
  "bedrooms": 2,
  "max_adults": 6,
  "price": 757,
  "goods": [
    `Coffee machine`,
    `Baby seat`,
    `Dishwasher`,
    `Fridge`,
    `Washing machine`,
    `Breakfast`,
  ],
  "host": {
    "id": 25,
    "name": `Angelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  "location": {
    "latitude": 50.945361,
    "longitude": 6.935974,
    "zoom": 16
  },
  "id": 1
};
