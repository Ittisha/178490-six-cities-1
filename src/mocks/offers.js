export const OFFERS = [
  {
    isPremium: true,
    isInBookmarks: false,
    title: `Beautiful & luxurious apartment at great location`,
    photoUrl: `img/apartment-01.jpg`,
    price: 120,
    type: `Apartment`,
    rating: `93%`,
    id: 1,
    coordinates: [52.3909553943508, 4.85309666406198],
    cityName: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
  },
  {
    isPremium: false,
    isInBookmarks: true,
    title: `Wood and stone place`,
    photoUrl: `img/room.jpg`,
    price: 80,
    type: `Private room`,
    rating: `80%`,
    id: 2,
    coordinates: [52.369553943508, 4.85309666406198],
    cityName: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
  },
  {
    isPremium: false,
    isInBookmarks: false,
    title: `Canal View Prinsengracht`,
    photoUrl: `img/apartment-02.jpg`,
    price: 132,
    type: `Apartment`,
    rating: `80%`,
    id: 3,
    coordinates: [52.3909553943508, 4.929309666406198],
    cityName: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
  },
  {
    isPremium: false,
    isInBookmarks: false,
    title: `Nice, cozy, warm big bed apartment`,
    photoUrl: `img/apartment-03.jpg`,
    price: 180,
    type: `Apartment`,
    rating: `80%`,
    id: 4,
    coordinates: [52.3809553943508, 4.939309666406198],
    cityName: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
  },
  {
    isPremium: false,
    isInBookmarks: false,
    title: `Nice, cozy, warm big bed apartment`,
    photoUrl: `img/apartment-03.jpg`,
    price: 100,
    type: `Apartment`,
    rating: `90%`,
    id: 5,
    coordinates: [48.866790, 2.378520],
    cityName: `Paris`,
    cityCoords: [48.864716, 2.349014],
  }

];
