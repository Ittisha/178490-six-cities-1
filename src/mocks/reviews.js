export const MOCK_REVIEWS = [
  {
    id: 1,
    user: {
      avatarUrl: ``,
      id: 1,
      isPro: false,
      name: ``,
    },
    rating: 4,
    comment: ``,
    date: ``,
  },
  {
    id: 2,
    user: {
      avatarUrl: ``,
      id: 5,
      isPro: false,
      name: ``,
    },
    rating: 3,
    comment: ``,
    date: ``,
  }
];

export const MOCK_REVIEW = {
  comment: ``,
  id: 1,
  rating: `3`,
};
