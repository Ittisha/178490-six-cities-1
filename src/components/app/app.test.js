import * as React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {App} from './app';
import {OFFERS} from '../../mocks/offers';
import {INIT_CITY} from '../../mocks/init-city';

configure({adapter: new Adapter()});

it(`App is rendered properly`, () => {
  const tree = shallow(
      <App
        offers={OFFERS}
        city={INIT_CITY}
        onAppMounting={() => {}}
      />);

  expect(tree).toMatchSnapshot();
});
