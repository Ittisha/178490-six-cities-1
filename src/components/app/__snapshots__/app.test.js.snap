// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is rendered properly 1`] = `ShallowWrapper {}`;
